### ISSUE TITLE
Xicoy X-range support

### ISSUE THREAD

**Issue by henke2** [thomasekdahlN/jeti#11]
Can we team up and get this working with the new Xicoy turbines using the built in ECU with everything brushless?

The scrips connects but he values seems way off....  I'm happy to help

**Comment by coretrekthomas** [comment]
Absolutely. I have a LUA script you can run to retrieve the parameters received from the turbine, so I can make a setup.

Run this script on the transmitter with the model selected and sensors discovered, and send me the file it creates.

https://github.com/thomasekdahlN/jeti/blob/master/senslist.lua

**Comment by henke2** [comment]
I downloaded the rawfile to my Tx named senslist.lua, I get inte application under user applications list but nothing seems to happen when I click it. No file being saved from what I can se. What am I missing?

**Comment by henke2** [comment]
Possibly this is it?

[REX12A.txt](https://github.com/thomasekdahlN/jeti/files/10773928/REX12A.txt)
[Turbine.txt](https://github.com/thomasekdahlN/jeti/files/10773929/Turbine.txt)

**Comment by henke2** [comment]
I added XicoyX folders and files and I'm actually getting it to worksort of. The values read correct now.

How ever I can't change tank size in the ECU dashboard on the DS24 and i get messages unknown status configuration on the display.

Getting there 👍

**Comment by henke2** [comment]
Found some numbers that might be the ones, just sticking them here until I have time to do the above two files and try.

Thanks for your help so far.

" [0] = ""TempHigh"",
[1] = ""Trim Low"",
[2] = ""SetIdle!"",
[3] = ""Ready"",
[4]= ""Ignition"",
[5] = ""FuelRamp"",
[6] = ""Glow Test"",
[7] = ""Running"",
[8] = ""Stop"",
[9] = ""FlameOut"",
[10] = ""SpeedLow"",
[11] = ""Cooling"",
[12] = ""Ignitor Bad"",
[13] = ""Starter Bad"",
[14] = ""Weak Gas"",
[15] = ""Start On"",
[16] = ""User Off"",
[17] = ""Failsafe"",
[18] = ""Low RPM"",
[19] = ""Reset"",
[20] = ""Rx PwFail"",
[21] = ""Pre Heat"",
[22] = ""Battery!"",
[23] = ""Time Out"",
[24] = ""Overload"",
[25] = ""Ign.Fail"",
[26] = ""BurnerOn"",
[27] = ""Starting"",
[28] = ""SwitchOv"",
[29] = ""Cal.Pump"",
[30] = ""Pump Limit"",
[31] = ""NoEngine"",
[32] = ""PwrBoost"",
[33] = ""Run-Idle"",
[34] = ""Run-Max "",
[35] = ""Restart "","

**Comment by henke2** [comment]
> Great. I can add the files you created to the official release.
> 
> The message unknown status configuration is probably due to the need to map turbine statuses, that it receives an unmapped code. Maybe you can see the unmapped code in the console log but the best is to get a list of all the valid turbine statuses from Xicoy. (Can you ask them?) It needs to be mapped in a new xicoy config file similar to this one:
> 
> https://github.com/thomasekdahlN/jeti/blob/master/ecu/converter/xicoy/xicoyv10/config.jsn
> 
> "statusmap" : { "0" : "Trim low", "1" : "Ready", "2" : "Stick low", "3" : "Glow test", "4" : "Start on", "5" : "Ignition", "6" : "Preheat", "7" : "Fuel ramp", "8" : "Running", "9" : "Stop", "10" : "Cooling", "11" : "Glow bad", "12" : "Start bad", "13" : "Low rpm", "14" : "High temp", "15" : "Flameout", "16" : "Set idle", "17" : "Burner on", "18" : "Start on", "19" : "Ignition failure", "20" : "Ignitor bad", "21" : "Battery low", "22" : "Unknown" }
> 
> It also needs a matching advanced setup like this file:
> 
> https://github.com/thomasekdahlN/jeti/blob/master/ecu/status/xicoyv10.jsn
> 
> { "sensorname": "status", "Device delay" : { "text" : "Device delay", "message" : { "enable" : true, "seconds": 5 }, "audio" : { "enable" : false, "repeat" : 1, "file" : "Device delay.wav" }, "haptic" : { "enable" : true, "repeat" : 1, "stick" : "left", "vibrationProfile" : 4 } }, "Flameout" : { "text" : "Flameout", "message" : { "enable" : true, "seconds": 5 }, "audio" : { "enable" : true, "repeat" : 1, "file" : "Flameout.wav" }, "haptic" : { "enable" : true, "repeat" : 1, "stick" : "left", "vibrationProfile" : 4 } },

I can't se how the xicoyv10.jsn matches the statuses in the config.jsn. They are not the same when I compare. Is the xicoyv10.jsn a generic file talking to the Lua app rather than to the telemetry adapter?

Also there are alot of audiofiles mixed, man and woman voices. Are this some sort of generic library or why this mix-up of different qualities? Wouldn't one set in the same voice make sense?   I'm new to this, so just wondering.

**Comment by henke2** [comment]
This is what I came up with. My first very basic tests seems to work. How ever I'm not really sure how it's supposed to work. Will provide a video later too.

[X45Tp.txt](https://github.com/thomasekdahlN/jeti/files/10837506/X45Tp.txt)
[xicoyX.txt](https://github.com/thomasekdahlN/jeti/files/10837510/xicoyX.txt)
[config.txt](https://github.com/thomasekdahlN/jeti/files/10837514/config.txt)

**Comment by natxopedreira** [comment]
I also have a new xicoy telemetry converter that label TelemetryX and im trying to use it on my ds14 with my x45, however im getting an error blaming on line 403 of lua_16

`attemp to index value (field"?")`

**Comment by thomasekdahlN** [comment]
I have released a new version, you can try it. It has to be installed from the git repo according to the readme, as it is not in the Jeti App store.
